core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -97,4 +111,34 @@ private ProbeResult fromSCMLink(ProbeContext context, String scm) {
     public String[] getProbeResultRequirement() {
         return new String[]{UpdateCenterPluginPublicationProbe.KEY};
     }
+
+    /**
+     * Searches for Pom files in every directory available in the repository
+     *
+     * @param directory  in the scm
+     * @param pluginName the name of the plugin
+     * @param scm        the valid scm link
+     * @return folderPath if it valid or return the scm itself
+     */
+    private String searchPomFiles(Path directory, String pluginName, String scm) {
+        MavenXpp3Reader mavenReader = new MavenXpp3Reader();
+
+        try (Stream<Path> paths = Files.find(directory, 1, (path, $) ->
+            path.getFileName().toString().equals("pom.xml")
+        )) {
+            try (Reader reader = new InputStreamReader(new FileInputStream((File) paths), StandardCharsets.UTF_8)) {
+                Model model = mavenReader.read(reader);
+                if (model.getPackaging().equals("hpi") && model.getArtifactId().equals(pluginName)) {
+                    return directory.toString();
+                }
```

Comment:
In order to ease writing tests, you can extract this to a dedicated function called something like `boolean pomFileMatchesPlugin(Path pomFile, String pluginName)` that will do the `pom.xml` parsing logic and tell you if the `pom.xml` file you provided indeed matches the plugin you are searching for.

Then you can simply use it with `paths.filter(pom -> pomFileMatchesPlugin(pom, pluginName))` which should contain a single entry.

From there, you can retrieve the directory you are searching for.